lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -254,6 +254,11 @@ static void addMandatoryDiagnosticOptPipeline(SILPassPipelinePlan &P) {
   P.addMandatoryPerformanceOptimizations();
   P.addOnoneSimplification();
   P.addInitializeStaticGlobals();
+
+  if (P.getOptions().EmbeddedSwift) {
+    P.addDeadFunctionAndGlobalElimination();
+  }
```

Comment:
Will it be useful to run this pass in non-embedded swift as well for performance annotations ?

Reply:
I think we don't want to run this pass in regular Swift in -Onone, because for debugging and development purposes, users expect that even unused functions are available in debug builds (for example to be called from the debugger).